loki/transformations/tests/test_array_indexing.py

```diff
@@ -1181,44 +1181,54 @@ def test_transform_resolve_vector_notation_common_loops(tmp_path, frontend):
 
 @pytest.mark.parametrize('frontend', available_frontends())
 @pytest.mark.parametrize('calls_only', (False, True))
-def test_transform_explicit_dimensions(tmp_path, frontend, builder, calls_only):
+@pytest.mark.parametrize('as_kwargs', (False, True))
+def test_transform_explicit_dimensions(tmp_path, frontend, builder, calls_only, as_kwargs):
     """
-    Test flattening or arrays, meaning converting multi-dimensional
-    arrays to one-dimensional arrays including corresponding
-    index arithmetic (for calls).
+    Test making dimensions of arrays explicit and undoing this,
+    thus removing colon notation from array dimensions either for all
+    or for arrays within (inline) calls only.
     """
-    fcode_driver = """
+    fcode_driver = f"""
   SUBROUTINE driver_routine(nlon, nlev, a, b)
-    use kernel_mod, only: kernel_routine
+    use kernel_explicit_dimensions_mod, only: kernel_routine
     INTEGER, INTENT(IN)    :: nlon, nlev
     INTEGER, INTENT(INOUT) :: a(nlon,nlev)
     INTEGER, INTENT(INOUT)  :: b(nlon,nlev)
 
-    call kernel_routine(nlon, nlev, a, b)
+    call kernel_routine(nlon, nlev, {'a=' if as_kwargs else ''}a, {'b=' if as_kwargs else ''}b)
```

Comment:
[no action] Just a minor nitpick: I don't think the additional parametrize and string-injection is needed. You could just leave `a` as argument and make `b` a keyword argument, but this way works too.